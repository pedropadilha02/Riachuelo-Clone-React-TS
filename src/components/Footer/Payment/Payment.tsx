import styled from 'styled-components';

const PaymentSection = styled.section`
  margin: 15px auto;
  width: 90%;

  @media (max-width: 1200px) {
    width: 100%;
    margin: 0;
  }
`;

const PaymentContainer = styled.div`
  display: flex;
  margin: 0 auto;
  width: 80%;
`;

const PaymentTitle = styled.h3`
  text-transform: uppercase;
  margin-bottom: 10px;
  display: block;
  font-size: 1em;
  font-weight: bold;
`;
const PaymentCards = styled.div`
  width: 50%;
  display: block;
`;

const PaymentImg = styled.img`
  margin: 2px;
  width: auto;
`;
const PaymentSocial = styled.div`
  flex-grow: 1;
  display: block;
`;

const SocialItemsContainer = styled.div`
  display: flex;
  justify-content: space-between;
  margin-right: 50px;
`;

const SocialItem = styled.div`
  display: block;
  margin: 0px;
  flex-wrap: wrap;
  box-sizing: border-box;
`;

const SocialText = styled.p`
  color: #261e1e;
  font-size: 11px;
  text-align: center;
`;

const SocialIcon = styled.img`
  display: block;
  margin: auto;
  height: 25px;
  width: 25px;
`;

const Accessibility = styled.div`
  flex-grow: 1;
  display: block;
  margin-left: 50px;
`;

const AccessibilityImg = styled.img`
  height: 30px;
  widht: 30px;
`;

export function Payment() {
  return (
    <PaymentSection>
      <PaymentContainer>
        <PaymentCards>
          <PaymentTitle>Formas de Pagamento</PaymentTitle>
          <PaymentImg
            src="http://localhost:3000/assets/ico-cartao-rchlo1.webp"
            alt=""
          />
          <PaymentImg
            src="http://localhost:3000/assets/ico-cartao-rchlo2.webp"
            alt=""
          />
          <PaymentImg
            src="http://localhost:3000/assets/ico-cartao-rchlo3.webp"
            alt=""
          />
          <PaymentImg
            src="http://localhost:3000/assets/ico-cartao-visa.png"
            alt=""
          />
          <PaymentImg
            src="http://localhost:3000/assets/ico-cartao-master.webp"
            alt=""
          />
          <PaymentImg
            src="http://localhost:3000/assets/ico-cartao-amex.webp"
            alt=""
          />
          <PaymentImg
            src="http://localhost:3000/assets/ico-cartao-diners.png"
            alt=""
          />
          <PaymentImg
            src="http://localhost:3000/assets/ico-cartao-elo.png"
            alt=""
          />
          <PaymentImg
            src="http://localhost:3000/assets/ico-boleto.webp"
            alt=""
          />
          <PaymentImg
            src="http://localhost:3000/assets/ico-paypal.png"
            alt=""
          />
        </PaymentCards>
        <PaymentSocial>
          <PaymentTitle>Siga a Riachuelo</PaymentTitle>
          <SocialItemsContainer>
            <SocialItem>
              <a href="#">
                <SocialIcon
                  src="http://localhost:3000/assets/ico-facebook.svg"
                  alt=""
                />
              </a>
            </SocialItem>

            <SocialItem>
              <a href="#">
                <SocialIcon
                  src="http://localhost:3000/assets/ico-instagram.svg"
                  alt=""
                />
              </a>
              <SocialText>RCHLO</SocialText>
            </SocialItem>

            <SocialItem>
              <a href="#">
                <SocialIcon
                  src="http://localhost:3000/assets/ico-instagram.svg"
                  alt=""
                />
              </a>
              <SocialText>
                CASA
                <br />
                RCHLO
              </SocialText>
            </SocialItem>

            <SocialItem>
              <a href="#">
                <SocialIcon
                  src="http://localhost:3000/assets/ico-pinterest.svg"
                  alt=""
                />
              </a>
            </SocialItem>
            <SocialItem>
              <a href="#">
                <SocialIcon
                  src="http://localhost:3000/assets/ico-twitter.svg"
                  alt=""
                />
              </a>
            </SocialItem>
            <SocialItem>
              <a href="#">
                <SocialIcon
                  src="http://localhost:3000/assets/ico-youtube.svg"
                  alt=""
                />
              </a>
            </SocialItem>
            <SocialItem>
              <a href="#">
                <SocialIcon
                  src="http://localhost:3000/assets/ico-linkedin.svg"
                  alt=""
                />
              </a>
            </SocialItem>
          </SocialItemsContainer>
        </PaymentSocial>
        <Accessibility>
          <PaymentTitle>Acessibilidade</PaymentTitle>
          <AccessibilityImg
            src="http://localhost:3000/assets/ico-a11y.webp"
            alt=""
          />
        </Accessibility>
      </PaymentContainer>
    </PaymentSection>
  );
}
