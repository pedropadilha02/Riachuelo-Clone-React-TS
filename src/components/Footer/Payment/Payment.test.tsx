import { render } from '@testing-library/react';
import { Payment } from './Payment';

describe('Teste de renderização do Payment', () => {
  it('O snapshot do component deve permanecer semre o mesmo', () => {
    const { container } = render(<Payment />);

    expect(container).toMatchSnapshot();
  });
});
