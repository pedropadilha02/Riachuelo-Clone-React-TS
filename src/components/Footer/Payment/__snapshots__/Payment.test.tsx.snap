// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de renderização do Payment O snapshot do component deve permanecer semre o mesmo 1`] = `
<div>
  <section
    class="sc-bdnxRM ccRLLJ"
  >
    <div
      class="sc-gtsrHT jzvizc"
    >
      <div
        class="sc-hKFxyN kmVPMd"
      >
        <h3
          class="sc-dlnjwi dRMNSy"
        >
          Formas de Pagamento
        </h3>
        <img
          alt=""
          class="sc-eCApnc jmKjNN"
          src="http://localhost:3000/assets/ico-cartao-rchlo1.webp"
        />
        <img
          alt=""
          class="sc-eCApnc jmKjNN"
          src="http://localhost:3000/assets/ico-cartao-rchlo2.webp"
        />
        <img
          alt=""
          class="sc-eCApnc jmKjNN"
          src="http://localhost:3000/assets/ico-cartao-rchlo3.webp"
        />
        <img
          alt=""
          class="sc-eCApnc jmKjNN"
          src="http://localhost:3000/assets/ico-cartao-visa.png"
        />
        <img
          alt=""
          class="sc-eCApnc jmKjNN"
          src="http://localhost:3000/assets/ico-cartao-master.webp"
        />
        <img
          alt=""
          class="sc-eCApnc jmKjNN"
          src="http://localhost:3000/assets/ico-cartao-amex.webp"
        />
        <img
          alt=""
          class="sc-eCApnc jmKjNN"
          src="http://localhost:3000/assets/ico-cartao-diners.png"
        />
        <img
          alt=""
          class="sc-eCApnc jmKjNN"
          src="http://localhost:3000/assets/ico-cartao-elo.png"
        />
        <img
          alt=""
          class="sc-eCApnc jmKjNN"
          src="http://localhost:3000/assets/ico-boleto.webp"
        />
        <img
          alt=""
          class="sc-eCApnc jmKjNN"
          src="http://localhost:3000/assets/ico-paypal.png"
        />
      </div>
      <div
        class="sc-jSFjdj hOEhMY"
      >
        <h3
          class="sc-dlnjwi dRMNSy"
        >
          Siga a Riachuelo
        </h3>
        <div
          class="sc-gKAaRy bZNrck"
        >
          <div
            class="sc-iCoGMd cSVfke"
          >
            <a
              href="#"
            >
              <img
                alt=""
                class="sc-pNWdM yVHFZ"
                src="http://localhost:3000/assets/ico-facebook.svg"
              />
            </a>
          </div>
          <div
            class="sc-iCoGMd cSVfke"
          >
            <a
              href="#"
            >
              <img
                alt=""
                class="sc-pNWdM yVHFZ"
                src="http://localhost:3000/assets/ico-instagram.svg"
              />
            </a>
            <p
              class="sc-fujyAs hGvzSs"
            >
              RCHLO
            </p>
          </div>
          <div
            class="sc-iCoGMd cSVfke"
          >
            <a
              href="#"
            >
              <img
                alt=""
                class="sc-pNWdM yVHFZ"
                src="http://localhost:3000/assets/ico-instagram.svg"
              />
            </a>
            <p
              class="sc-fujyAs hGvzSs"
            >
              CASA
              <br />
              RCHLO
            </p>
          </div>
          <div
            class="sc-iCoGMd cSVfke"
          >
            <a
              href="#"
            >
              <img
                alt=""
                class="sc-pNWdM yVHFZ"
                src="http://localhost:3000/assets/ico-pinterest.svg"
              />
            </a>
          </div>
          <div
            class="sc-iCoGMd cSVfke"
          >
            <a
              href="#"
            >
              <img
                alt=""
                class="sc-pNWdM yVHFZ"
                src="http://localhost:3000/assets/ico-twitter.svg"
              />
            </a>
          </div>
          <div
            class="sc-iCoGMd cSVfke"
          >
            <a
              href="#"
            >
              <img
                alt=""
                class="sc-pNWdM yVHFZ"
                src="http://localhost:3000/assets/ico-youtube.svg"
              />
            </a>
          </div>
          <div
            class="sc-iCoGMd cSVfke"
          >
            <a
              href="#"
            >
              <img
                alt=""
                class="sc-pNWdM yVHFZ"
                src="http://localhost:3000/assets/ico-linkedin.svg"
              />
            </a>
          </div>
        </div>
      </div>
      <div
        class="sc-jrsJWt eBlIzz"
      >
        <h3
          class="sc-dlnjwi dRMNSy"
        >
          Acessibilidade
        </h3>
        <img
          alt=""
          class="sc-kEqXSa kWsKLu"
          src="http://localhost:3000/assets/ico-a11y.webp"
        />
      </div>
    </div>
  </section>
</div>
`;
